proto/beacon/p2p/v1/types.proto

```diff
@@ -46,6 +46,52 @@ message BeaconState {
   ethereum.eth.v1alpha1.Checkpoint finalized_checkpoint = 8004;
 }
 
+// Beacon state for hard fork 1.
+// https://github.com/ethereum/eth2.0-specs/blob/dev/specs/lightclient/beacon-chain.md#beaconstate
+message BeaconStateV1 {
+  // Versioning [1001-2000]
+  uint64 genesis_time = 1001;
+  bytes genesis_validators_root = 1002 [(gogoproto.moretags) = "ssz-size:\"32\""];
+  uint64 slot = 1003 [(gogoproto.casttype) = "github.com/prysmaticlabs/eth2-types.Slot"];
+  Fork fork = 1004;
+
+  // History [2001-3000]
+  ethereum.eth.v1alpha1.BeaconBlockHeader latest_block_header = 2001;
+  repeated bytes block_roots = 2002 [(gogoproto.moretags) = "ssz-size:\"block_roots.size\""];
+  repeated bytes state_roots = 2003 [(gogoproto.moretags) = "ssz-size:\"state_roots.size\""];
+  repeated bytes historical_roots = 2004 [(gogoproto.moretags) = "ssz-size:\"?,32\" ssz-max:\"16777216\""];
+
+  // Eth1 [3001-4000]
+  ethereum.eth.v1alpha1.Eth1Data eth1_data = 3001;
+  repeated ethereum.eth.v1alpha1.Eth1Data eth1_data_votes = 3002 [(gogoproto.moretags) = "ssz-max:\"eth1_data_votes.size\""];
+  uint64 eth1_deposit_index = 3003;
+
+  // Registry [4001-5000]
+  repeated ethereum.eth.v1alpha1.Validator validators = 4001 [(gogoproto.moretags) = "ssz-max:\"1099511627776\""];
+  repeated uint64 balances = 4002 [(gogoproto.moretags) = "ssz-max:\"1099511627776\""];
+
+  // Randomness [5001-6000]
+  repeated bytes randao_mixes = 5001 [(gogoproto.moretags) = "ssz-size:\"randao_mixes.size\""];
+
+  // Slashings [6001-7000]
+  repeated uint64 slashings = 6001 [(gogoproto.moretags) = "ssz-size:\"slashings.size\""];
+
+  // Participation [7001-8000]
+  bytes previous_epoch_participation = 7001 [(gogoproto.moretags) = "ssz-max:\"1099511627776\""];;
+  bytes current_epoch_participation = 7002 [(gogoproto.moretags) = "ssz-max:\"1099511627776\""];;
```

Comment:
Both should be `repeated bytes`

Reply:
It's actually `bytes` by design since each validator represents 1 byte of data.

Generated code:
```go
PreviousEpochParticipation  []byte                                          `protobuf:"bytes,7001,opt,name=previous_epoch_participation,json=previousEpochParticipation,proto3" json:"previous_epoch_participation,omitempty" ssz-max:"1099511627776"`
	CurrentEpochParticipation   []byte                                          `protobuf:"bytes,7002,opt,name=current_epoch_participation,json=currentEpochParticipation,proto3" json:"current_epoch_participation,omitempty" ssz-max:"1099511627776"`
```